### PR TITLE
apparmor: Add apparmor profile for postgresql

### PR DESCRIPTION
**Pull request by Lakshmipathi**
Add apparmor profile in the postgres ami image.

Signed-off-by: Lakshmipathi Ganapathi <lakshmi@supabase.io>

### CODE REVIEW

**Comment by Lakshmipathi** [comment]
@darora @dragarcia @inian Draft with postgres profile. Need to add pgbouncer profile too.

File: ebssurrogate/files/apparmor_profiles/usr.bin.vector [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  #include <abstractions/totem>
+  #include <abstractions/user-tmp>
+
+  deny @{HOME}/** rwx,
+  /etc/vector/** r,
+  /run/systemd/notify rw,
+  /sys/kernel/mm/transparent_hugepage/enabled r,
+  /usr/bin/vector mrix,
+  /var/lib/vector/** rw,
+  /var/log/postgresql/** rw,
```
Comment (darora) [inline ebssurrogate/files/apparmor_profiles/usr.bin.vector]: write needed?

File: ebssurrogate/files/apparmor_profiles/usr.lib.postgresql.bin.postgres [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+/dev/shm rw,
+deny @{HOME}/** rwx,
+/etc/postgresql-custom/** r,
+/etc/postgresql/** r,
+/run/systemd/notify rw,
+/var/run/systemd/notify rw,
+/usr/bin/cat rix,
+/usr/bin/dash rix,
+/usr/bin/mknod rix,
+/usr/lib/postgresql/bin/* mrwix,
```
Comment (darora) [inline ebssurrogate/files/apparmor_profiles/usr.lib.postgresql.bin.postgres]: write needed?

File: ebssurrogate/files/apparmor_profiles/usr.local.bin.pgbouncer [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  #include <abstractions/openssl>
+  #include <abstractions/ssl_keys>
+  #include <abstractions/user-tmp>
+
+  deny @{HOME}/** rwx,
+  /etc/pgbouncer-custom/** r,
+  /etc/pgbouncer/** r,
+  /proc/sys/kernel/random/uuid r,
+  /run/systemd/notify rw,
+  /usr/local/bin/pgbouncer mrwix,
```
Comment (darora) [inline ebssurrogate/files/apparmor_profiles/usr.local.bin.pgbouncer]: w?

File: ebssurrogate/files/apparmor_profiles/usr.lib.postgresql.bin.postgres (hunk 2 above)
Comment (darora) [reply ebssurrogate/files/apparmor_profiles/usr.lib.postgresql.bin.postgres]: (at a more general level, trying to apply w^x)

File: ebssurrogate/files/apparmor_profiles/usr.local.bin.pgbouncer (hunk 3 above)
Comment (Lakshmipathi) [reply ebssurrogate/files/apparmor_profiles/usr.local.bin.pgbouncer]: no,it shouldn't be required.

File: ebssurrogate/files/apparmor_profiles/usr.lib.postgresql.bin.postgres (hunk 2 above)
Comment (Lakshmipathi) [reply ebssurrogate/files/apparmor_profiles/usr.lib.postgresql.bin.postgres]: let me remove this too.

**Review by darora — APPROVED** [review]

**Pull request merged**